pkg/gbpserver/watchers/netflow_watch.go

```diff
@@ -0,0 +1,205 @@
+	nfw.watchNetflow(stopCh)
+	return nil
+}
+
+func (nfw *NetflowWatcher) watchNetflow(stopCh <-chan struct{}) {
+
+	NetflowLw := cache.NewListWatchFromClient(nfw.rc, "netflowpolicies", metav1.NamespaceAll, fields.Everything())
+	_, netflowInformer := cache.NewInformer(NetflowLw, &netflowpolicy.NetflowPolicy{}, 0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				nfw.log.Infof("netflow added")
+				nfw.netflowAdded(obj)
+			},
+			UpdateFunc: func(oldobj interface{}, newobj interface{}) {
+				nfw.log.Infof("netflow updated")
+				nfw.netflowAdded(newobj)
+			},
+			DeleteFunc: func(obj interface{}) {
+				nfw.log.Infof("netflow deleted")
+				nfw.netflowDeleted(obj)
+			},
+		})
+	go netflowInformer.Run(stopCh)
+}
+
+func (nfw *NetflowWatcher) netflowAdded(obj interface{}) {
+	netflow, ok := obj.(*netflowpolicy.NetflowPolicy)
+	if !ok {
+		nfw.log.Errorf("netflowAdded: Bad object type")
+		return
+	}
+
+	nfw.log.Infof("netflowAdded - %s", netflow.ObjectMeta.Name)
+	netflowMO := &netflowCRD{
+		DstAddr:           netflow.Spec.FlowSamplingPolicy.DstAddr,
+		DstPort:           netflow.Spec.FlowSamplingPolicy.DstPort,
+		Version:           netflow.Spec.FlowSamplingPolicy.Version,
+		ActiveFlowTimeOut: netflow.Spec.FlowSamplingPolicy.ActiveFlowTimeOut,
+		IdleFlowTimeOut:   netflow.Spec.FlowSamplingPolicy.IdleFlowTimeOut,
+		SamplingRate:      netflow.Spec.FlowSamplingPolicy.SamplingRate,
+		Name:              netflow.ObjectMeta.Name,
+	}
+	nfw.gs.AddGBPCustomMo(netflowMO)
+}
+
+func (nfw *NetflowWatcher) netflowDeleted(obj interface{}) {
+	netflow, ok := obj.(*netflowpolicy.NetflowPolicy)
+	if !ok {
+		nfw.log.Errorf("netflowDeleted: Bad object type")
+		return
+	}
+
+	nfw.log.Infof("netflowDeleted - %s", netflow.ObjectMeta.Name)
+	netflowMO := &netflowCRD{
+		DstAddr:           netflow.Spec.FlowSamplingPolicy.DstAddr,
+		DstPort:           netflow.Spec.FlowSamplingPolicy.DstPort,
+		Version:           netflow.Spec.FlowSamplingPolicy.Version,
+		ActiveFlowTimeOut: netflow.Spec.FlowSamplingPolicy.ActiveFlowTimeOut,
+		IdleFlowTimeOut:   netflow.Spec.FlowSamplingPolicy.IdleFlowTimeOut,
+		SamplingRate:      netflow.Spec.FlowSamplingPolicy.SamplingRate,
+		Name:              netflow.ObjectMeta.Name,
+	}
+	nfw.gs.DelGBPCustomMo(netflowMO)
+}
+
+func (nf *netflowCRD) fillNetflowDefaults() {
+
+	// setting default values if no values present for netflow attributes.
+	if nf.Version == "netflow" {
+		nf.Version = "v5"
+	} else if nf.Version == "ipfix" {
+		nf.Version = "v9"
+	} else {
+		nf.Version = "v5"
+	}
+	if nf.DstPort == 0 {
+		nf.DstPort = 2055
+	}
+	if nf.ActiveFlowTimeOut == 0 {
+		nf.ActiveFlowTimeOut = 60
+	}
+	if nf.IdleFlowTimeOut == 0 {
+		nf.IdleFlowTimeOut = 15
+	}
+	if nf.SamplingRate == 0 {
+		nf.SamplingRate = 0
+	}
+}
+
+func (nf *netflowCRD) Subject() string {
+	return netflowCRDSub
+}
+
+func (nf *netflowCRD) URI() string {
+	gServer := &gbpserver.Server{}
```

Comment:
This doesn't seem like the right way to get an initialized gbpserver.Server instance

Reply:
working on this